### PR TITLE
Bug 1587974  - Catch error when retrieving profile

### PR DESCRIPTION
**Pull request by helfi92**
Bugzilla Bug: [https://bugzilla.mozilla.org/show_bug.cgi?id=1587974](https://bugzilla.mozilla.org/show_bug.cgi?id=1587974)

Can't seem to reproduce but I think this is a possible fix.

### CODE REVIEW

File: services/web-server/src/login/strategies/mozilla-auth0.js [hunk 1]
```diff
@@ -62,9 +62,14 @@ module.exports = class MozillaAuth0 {
 
   async getUser({ userId }) {
     const personApi = await this.getPersonApi();
-    const userProfile = await personApi.getProfileFromUserId(userId);
+    const [userProfileErr, userProfile] = await tryCatch(personApi.getProfileFromUserId(userId));
     const user = new User();
 
+    if (userProfileErr) {
+      debug(`error retrieving profile from the PersonAPI`);
```
Comment (djmitche) [inline services/web-server/src/login/strategies/mozilla-auth0.js]: Let's report that error, too.  If "user not found" is treated as an error, then we could check for that condition and report any *other* errors.

File: services/web-server/src/login/strategies/mozilla-auth0.js [hunk 2]
```diff
@@ -192,9 +202,9 @@ module.exports = class MozillaAuth0 {
         // extraParams.id_token has the JSON Web Token
         // profile has all the information from the user
         async (accessToken, refreshToken, extraParams, profile, done) => {
```
Comment (djmitche) [inline services/web-server/src/login/strategies/mozilla-auth0.js]: Ah, so this is where it wasn't handled.  Nice find!

File: services/web-server/src/login/strategies/mozilla-auth0.js (hunk 1 above)
Comment (djmitche) [reply services/web-server/src/login/strategies/mozilla-auth0.js]: My thinking was to report everything *except* userNotFound -- we might want to see, for example, if a lot of our People API calls are 500'ing, a report in Sentry.  But if a bunch of people get fired and the client scanner doesn't find them when it looks them up, that's a "normal" failure and doesn't need to be reported.

The other issue with the arrangement as written here is that it treats a 500 from the people API as a missing user, which will cause us to disable clients.

So I think the right thing to do here is to propagate this error out of `getUser`, and then down where it says `// it is likely in the service logs` use `monitor.reportError` to put it in the service logs.

**Comment by helfi92** [comment]
> The other issue with the arrangement as written here is that it treats a 500 from the people API as a missing user, which will cause us to disable clients.

That's a really good observation!

File: services/web-server/src/login/strategies/github.js [hunk 3]
```diff
@@ -129,9 +131,16 @@ module.exports = class Github {
...
             accessToken,
           }, true);
-          const user = await this.getUser({ username: profile.username, userId: Number(profile.id) });
-
-          if (!user) {
+          const [userErr, user] = await tryCatch(
+            this.getUser({ username: profile.username, userId: Number(profile.id) })
+          );
+
+          if (userErr || !user) {
```
Comment (djmitche) [inline services/web-server/src/login/strategies/github.js]: I think we just want `if (userErr)` here -- the `!user` case is "normal operation"

File: services/web-server/src/login/strategies/mozilla-auth0.js [hunk 4]
```diff
@@ -192,9 +193,13 @@ module.exports = class MozillaAuth0 {
...
-          const user = await this.getUser({ userId: profile.user_id });
+          const [userErr, user] = await tryCatch(this.getUser({ userId: profile.user_id }));
 
-          if (!user) {
+          if (userErr || !user) {
```
Comment (djmitche) [inline services/web-server/src/login/strategies/mozilla-auth0.js]: same here

**Pull request merged**